napari/_vispy/canvas.py

```diff
@@ -38,26 +119,73 @@ def __init__(self, *args, **kwargs) -> None:
         # using an lru_cache.
         self.max_texture_sizes = get_max_texture_sizes()
 
-        self.events.ignore_callback_errors = False
-        self.context.set_depth_func('lequal')
+        for overlay in self.viewer._overlays.values():
+            self._add_overlay_to_visual(overlay)
+
+        self._scene_canvas.events.ignore_callback_errors = False
+        self._scene_canvas.context.set_depth_func('lequal')
+
+        # Connecting events from SceneCanvas
+        self._scene_canvas.events.key_press.connect(
+            self._key_map_handler.on_key_press
+        )
+        self._scene_canvas.events.key_release.connect(
+            self._key_map_handler.on_key_release
+        )
+        self._scene_canvas.events.draw.connect(self.enable_dims_play)
+        self._scene_canvas.events.draw.connect(self.vispy_camera.on_draw)
+
+        self._scene_canvas.events.mouse_double_click.connect(
+            self._on_mouse_double_click
+        )
+        self._scene_canvas.events.mouse_move.connect(self._on_mouse_move)
+        self._scene_canvas.events.mouse_press.connect(self._on_mouse_press)
+        self._scene_canvas.events.mouse_release.connect(self._on_mouse_release)
+        self._scene_canvas.events.mouse_wheel.connect(self._on_mouse_wheel)
+        self._scene_canvas.events.resize.connect(self.on_resize)
+        self._scene_canvas.events.draw.connect(self.on_draw)
+        self.viewer.cursor.events.style.connect(self._on_cursor)
+        self.viewer.cursor.events.size.connect(self._on_cursor)
+        self.viewer.events.theme.connect(self._on_theme_change)
+        self.viewer.camera.events.interactive.connect(self._on_interactive)
+        self.viewer.layers.events.reordered.connect(self._reorder_layers)
+        self.viewer.layers.events.removed.connect(self._remove_layer)
+        self.destroyed.connect(self._disconnect_theme)
+
+    @property
+    def destroyed(self) -> pyqtBoundSignal:
+        return self._scene_canvas._backend.destroyed
+
+    @property
+    def native(self):
+        """Returns the native widget of the Vispy SceneCanvas."""
+        return self._scene_canvas.native
 
     @property
-    def destroyed(self):
-        return self._backend.destroyed
+    def screen_changed(self):
+        """Returning signal indicating whether the window screen has changed."""
+        return self._scene_canvas._backend.screen_changed
 
     @property
-    def background_color_override(self):
-        return self._background_color_override
+    def background_color_override(self) -> str:
```

Comment:
```suggestion
    def background_color_override(self) -> Optional[str]:
```

Based on the current code and dostring.

Reply:
resolved with [74d4cbc](https://github.com/napari/napari/pull/5432/commits/74d4cbc129feea9b4ab36230328248480ff76224)